src/ensembl/src/content/app/entity-viewer/state/general/entityViewerGeneralActions.ts

```diff
@@ -94,8 +95,18 @@ export const setDataFromUrl: ActionCreator<ThunkAction<
       })
     : null;
 
-  if (entityId && entityId !== activeEntityId) {
-    dispatch(updateEnsObject(entityId));
+  if (entityId && genomeIdFromUrl) {
+    if (entityId !== activeEntityId) {
+      dispatch(updateEnsObject(entityId));
+    }
+    dispatch(setSidebarInitialStateForGenome(genomeIdFromUrl));
```

Comment:
Still think that there shouldn't be a dedicated action for this, but rather the sidebar reducer should do this when it receives new entity id. We may address this in the same PR as the cleaning up of ensObjects.